src/test/java/com/google/sps/ProfileBuilderServletTest.java

```diff
@@ -0,0 +1,266 @@
+    String userType = "Mentee";
+    when(request.getParameter("user-Type")).thenReturn(userType);
+    String menteeName = "Sandy Cane";
+    when(request.getParameter("name-input")).thenReturn(menteeName);
+    String school = "Stanford University";
+    when(request.getParameter("school-input")).thenReturn(school);
+    String major = "Computer Science";
+    when(request.getParameter("major-input")).thenReturn(major);
+
+    profileBuilderServlet.doPost(request, response);
+    PreparedQuery results = ds.prepare(new Query("User"));
+    String nameResult = "";
+    String schoolResult = "";
+    String majorResult = "";
+    for (Entity entity : results.asIterable()) {
+        nameResult = (String) entity.getProperty("name");
+        schoolResult = (String) entity.getProperty("school");
+        majorResult = (String) entity.getProperty("major");
+    }
+    Assert.assertEquals(menteeName, nameResult);
+    Assert.assertEquals(school, schoolResult);
+    Assert.assertEquals(major, majorResult);
+  }
+
+  @Test
+  public void checkMentorInfoIsSame() throws IOException, ServletException {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    String userType = "Mentor";
+    when(request.getParameter("user-Type")).thenReturn(userType);
+    String mentorName = "Jose Martinez";
+    when(request.getParameter("profName-input")).thenReturn(mentorName);
+    String company = "Google";
+    when(request.getParameter("company-input")).thenReturn(company);
+    String occupation = "Site Reliability Engineer";
+    when(request.getParameter("careerTitle-input")).thenReturn(occupation);
+    String[] specialties = new String[2];
+    specialties[0] = "Cloud Computing";
+    specialties[1] = "Infrastructure";
+    when(request.getParameterValues("specialty-input")).thenReturn(specialties);
+
+    profileBuilderServlet.doPost(request, response);
+    PreparedQuery results = ds.prepare(new Query("User"));
+    String nameResult = "";
+    String companyResult = "";
+    String occupationResult = "";
+    Collection<String> specialtiesCollection = new ArrayList<String>();
+    for (Entity entity : results.asIterable()) {
+        nameResult = (String) entity.getProperty("name");
+        companyResult = (String) entity.getProperty("company");
+        occupationResult = (String) entity.getProperty("occupation");  
+        specialtiesCollection = (Collection<String>) entity.getProperty("specialties");
+    }
+    String[] specialtiesResult = new String[specialtiesCollection.size()];
+    specialtiesResult = specialtiesCollection.toArray(specialtiesResult);
+    
+    Assert.assertEquals(mentorName, nameResult);
+    Assert.assertEquals(company, companyResult);
+    Assert.assertEquals(occupation, occupationResult);
+    Assert.assertArrayEquals(specialties, specialtiesResult);
+  }
+    
+    
+  @Test
+  public void checkMentorSpecialityIsSame() throws IOException, ServletException {
```

Comment:
[L220] Is this ('checkMentorSpecialityIsSame' method) redundant now that you have 'checkMentorInfoIsSame', which includes specialties? Or is there some other part of the logic you intended to test with it?
Same question for 'sendUserCheckIfNameMatches'.

Reply:
Yes It is. I removed both.

Reply:
Done